particle/particle/particle.py

```diff
@@ -694,6 +696,9 @@ def findall(cls, filter_fn=None, particle=None, **search_terms):
                     try:
                         if not filter_fn(item):
                             continue
+                    except NameError as err:  # catch what is undefined, e.g. a unit!
+                        raise NameError(err)
+                        break
```

Comment:
What exactly does this `break` do? And why re-raise the error (like that)? Wouldn't just not catching it just have the same effect?

Reply:
Oops, the `break` is a remnant :S. I will remove the line in a sec.

The specific catch of `NameError` is crucial to make sure undefined variables are flagged to the user rather than ignored. That is very different from other comparisons that are irrelevant, or should be ignored. Imagine I use a lambda function with a unit, but forget to import it - say `picosecond` from `hepunits`. The code should not silently treat that as a failing match ;-).

I'm not saying here that the whole logic in `findall` should not be reviewed. There may be optimisations/simplifications to implement. But my first-order goal here is to get the searches robust and correct.